### PR TITLE
Convert ResourceHandler to class (Removes ts-ignore from app options)

### PR DESCRIPTION
**Pull request by kpal81xd**
Fixes https://github.com/playcanvas/engine/pull/5986

This PR converts the ResourceHandler interface into a class and all implementations into child classes. This removes the need to add `@ts-ignore` to each Handler used in app options in examples

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by mvaligursky** [comment]
related to https://github.com/playcanvas/engine/pull/5574

File: src/framework/handlers/hierarchy.js
Comment (kungfooman) [inline src/framework/handlers/hierarchy.js:19]: Since `ResourceHandler#constructor` is setting `this._app` now, should we remove it in other places?

File: src/framework/handlers/scene-settings.js
Comment (kungfooman) [inline src/framework/handlers/scene-settings.js:11]: (here is another superfluous app assignment)

**Comment by kungfooman** [comment]
Looks good to me besides the mentioned superfluous app assignments which are also in:
- SceneHandler
- ScriptHandler
- TemplateHandler

**Comment by kpal81xd** [comment]
> Looks good to me besides the mentioned superfluous app assignments which are also in:
> 
> * SceneHandler
> * ScriptHandler
> * TemplateHandler

Just commit to fix all those rn 👍

**Review by kungfooman — APPROVED** [review]
:100:

**Review by slimbuck — APPROVED** [review]
Nice! (Pls get a second approval before merging).

**Pull request closed without merging**
